core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-reports/components/dot-experiments-report-daily-details/dot-experiments-report-daily-details.component.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator/jest';
+
+import { ConfirmationService } from 'primeng/api';
+
+import { DotMessageService } from '@dotcms/data-access';
+
+import { DotExperimentsReportDailyDetailsComponent } from './dot-experiments-report-daily-details.component';
+
+import { DotExperimentsDetailsTableComponent } from '../../../shared/ui/dot-experiments-details-table/dot-experiments-details-table.component';
+import { DotExperimentsReportsStore } from '../../store/dot-experiments-reports-store';
+
+describe('DotExperimentsReportDailyDetailsComponent', () => {
+    let spectator: Spectator<DotExperimentsReportDailyDetailsComponent>;
+    const createComponent = createComponentFactory({
+        component: DotExperimentsReportDailyDetailsComponent,
+        providers: [
+            mockProvider(ConfirmationService),
+            mockProvider(DotExperimentsReportsStore),
+            mockProvider(DotMessageService)
+        ]
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+    });
+
+    it('should create', () => {
+        expect(spectator.query(DotExperimentsDetailsTableComponent)).toExist();
```

Comment:
This need more testing no?

Reply:
Added an integration. If you click on Promote Button, show the `ConfirmPopup` and if you click ok, check if I called the `promoteVariant` of the store.

Reply:
Throw a comment here saying that, please.

Reply:
Added test.